mlflow/java/src/main/java/org/mlflow/sagemaker/ScoringServer.java

```diff
@@ -0,0 +1,270 @@
+import org.mlflow.models.Model;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+/** A RESTful webserver for {@link Predictor Predictors} that runs on the local host */
+public class ScoringServer {
+  public static final String RESPONSE_KEY_ERROR_MESSAGE = "Error";
+
+  public static final int HTTP_RESPONSE_CODE_SERVER_ERROR = 500;
+  public static final int HTTP_RESPONSE_CODE_SUCCESS = 200;
+  public static final int DEFAULT_PORT = 5001;
+
+  private enum RequestContentType {
+    Csv("text/csv"),
+    Json("application/json"),
+    Invalid("invalid");
+
+    private final String value;
+    private static final Map<String, RequestContentType> BY_VALUE_MAP = new HashMap<>();
+
+    static {
+      for (RequestContentType inputType : RequestContentType.values()) {
+        BY_VALUE_MAP.put(inputType.value, inputType);
+      }
+    }
+
+    RequestContentType(String value) {
+      this.value = value;
+    }
+
+    public static RequestContentType fromValue(String value) {
+      if (BY_VALUE_MAP.containsKey(value)) {
+        return BY_VALUE_MAP.get(value);
+      } else {
+        return RequestContentType.Invalid;
+      }
+    }
+  }
+
+  private final Optional<Predictor> predictor;
+  private final Optional<Integer> portNumber;
+  private Optional<Service> activeService = Optional.empty();
+
+  /**
+   * Constructs a {@link ScoringServer} to serve the specified {@link Predictor} on the local host
+   * at the default port: {@link ScoringServer#DEFAULT_PORT}
+   */
+  public ScoringServer(Predictor predictor) {
+    this(Optional.of(predictor), Optional.empty());
+  }
+
+  /**
+   * Constructs a {@link ScoringServer} to serve the specified {@link Predictor} on the local host
+   * at the specified port
+   */
+  public ScoringServer(Predictor predictor, int portNumber) {
+    this(Optional.of(predictor), Optional.of(portNumber));
+  }
+
+  /**
+   * Loads the MLFlow model at the specified path as a {@link Predictor} and serves it on the local
+   * host at the default port: {@link ScoringServer#DEFAULT_PORT}
+   *
+   * @param modelPath The path to the MLFlow model to serve
+   */
+  public ScoringServer(String modelPath) throws IOException, PredictorLoadingException {
+    this(modelPath, Optional.empty(), true);
+  }
+
+  /**
+   * Loads the MLFlow model at the specified path as a {@link Predictor} and serves it on the local
+   * host at the specified port
+   *
+   * @param modelPath The path to the MLFlow model to serve
+   */
+  public ScoringServer(String modelPath, int portNumber)
+      throws IOException, PredictorLoadingException {
+    this(modelPath, Optional.of(portNumber), true);
+  }
+
+  /**
+   * Loads the MLFlow model at the specified path as a {@link Predictor} and serves it on the local
+   * host at the specified port
+   *
+   * @param modelPath The path to the MLFlow model to serve
+   * @param failOnUnsuccessfulModelLoad If `true`, an exception will be thrown if the specified
```

Comment:
Let's just assume failOnUnsuccessfulModelLoad=true and remove the constructor.

Reply:
Removed!